### DISCUSSION TITLE
Create an edge with my own ID using a Sigma MultiGraph

### DISCUSSION THREAD

**Discussion by cjdubois-ptech** [jacomyal/sigma.js#1455] [Q&A] [answered]
I am visualizing some data using a sigma multigraph. When I create a node I am able to do something like this:
`sigmaInstance.graph.addNode(nodeId, node)`
On the other hand, when I am adding an edge, it looks like I have to do the following:
`sigmaInstance.graph.addEdge(sourceId, targetId, {edge})`
And when I add the edge, it looks like sigma is generating its own IDs for each edge.

Q: Is there a way that I can manually set the id for the edge that I am adding? either with a different implementation of addEdge or some other function or library that I am not aware of?

**Comment by sim51** [comment, marked as answer]
You should take a look at the graphology documentation, specially the `addEdgeWithKey` function : https://graphology.github.io/mutation.html#addedgewithkey

**Reply by cjdubois-ptech** [reply]
This worked for me. I overlooked the graphology docs and was looking only at the sigma documentation. Thank you for pointing me in the right direction.
